pkg/k8s/network_policy.go

```diff
@@ -264,9 +272,12 @@ func ipBlockToCIDRRule(block *slim_networkingv1.IPBlock) api.CIDRRule {
 }
 
 // parsePorts converts list of K8s NetworkPolicyPorts to Cilium PortRules.
-func parsePorts(ports []slim_networkingv1.NetworkPolicyPort) []api.PortRule {
+func parsePorts(ports []slim_networkingv1.NetworkPolicyPort) ([]api.PortRule, error) {
 	portRules := []api.PortRule{}
 	for _, port := range ports {
+		if port.EndPort != nil {
+			return portRules, ErrEndPortUnsupported
```

Comment:
Returning this error from here will cause this network policy parsing to be retried forever so the logs will be spammed with "Error while parsing k8s kubernetes NetworkPolicy".

Problem of course isn't really here, but in `K8sWatcher.addK8sNetworkPolicyV1`. It shouldn't be returning an error on parse failures since we don't want those to be forever retried.

Can we do something better than just log and forget here though? Would it make sense to set the NetworkPolicy's status to report that Cilium does not support this? Would a metric gauge be helpful to track how many policies are invalid?

Reply:
This has long been a pain point in validating network policies, and I agree it feels like we should be able to have a better solution. So far this is pretty much what we do (though I think usually it's in the `sanitize()` functions, not out here in the parse function).

Unfortunately updating networkpolicy status from arbitrary agents can introduce a scale issue by hammering the kube-apiserver from thousands of agents at once. We could consider moving some of this sort of validation into the operator, which I believe does not currently monitor for network policies. For a lot of the CNP validation we have been pushing it server-side into kube-apiserver... I'm not sure whether we could do that from Cilium for KNP as well, I think the validation is tied into resource registration. If we could change this for KNP, that would be the ideal solution --- prevent the user from even inserting the policy with these statements.

I like the metric idea.